## **Install libraries**

In [2]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
import urllib

## **Loading data**

In [3]:
url="https://web.greatervalleychamber.com/search"

In [ ]:
site = requests.get(url)
soup=BeautifulSoup(site.content,'lxml')
soup

## **Get categories**

In [5]:
all_category=[]
get_category =soup.find_all("select",{'id':"content1_Directory1_DirectorySearch1_SearchCategory_ddKeyword"})
for category in get_category:
    options=category.find_all("option")
    for option in options:
        all_category.append(option.text)

In [6]:
all_category=all_category[1:]
all_category
category=[]
for item in all_category:
    new=item.replace(' ','-')
    category.append(new)

In [7]:
len(category)

113

## **Fetch necessary data from the 113 options saving in Data Frame**

In [8]:
all_data = []
for name in category:
    url = f"https://web.greatervalleychamber.com/{name}"
    site = requests.get(url)
    print(name)
    print("Successful data extraction")
    soup = BeautifulSoup(site.content, 'lxml')
    b_name = soup.find_all("div", class_="ListingResults_All_ENTRYTITLELEFTBOX")
    b_info = soup.find_all("div", class_="ListingResults_Level1_MAINLEFTBOX")
    b_links = soup.find_all("div", "ListingResults_Level1_SOCIALMEDIA")

    for name, info, link in zip(b_name, b_info, b_links):
        get_a = link.find_all("a")
        links = []
        for a in get_a:
            unclean_link = a.get("href")
            extracted_url_part = unclean_link.split('URL=')[1].split("&")[0]
            url = urllib.parse.unquote(extracted_url_part)
            links.append(url)
        data = {"name": name.text.strip(), "info": info.text.strip(), "social": links}
        all_data.append(data)

Accounting-Services
Successful data extraction
Advertising
Successful data extraction
Agriculture
Successful data extraction
Animal-Hospital
Successful data extraction
Architects
Successful data extraction
Assisted-Living
Successful data extraction
Association
Successful data extraction
Attorney
Successful data extraction
Attraction
Successful data extraction
Audio-Visual-Service
Successful data extraction
Automotive
Successful data extraction
Aviation
Successful data extraction
Bakery
Successful data extraction
Banking-Services
Successful data extraction
Barter-Exchange
Successful data extraction
Beverages
Successful data extraction
Brewery
Successful data extraction
Business-Coaching
Successful data extraction
Business-Services
Successful data extraction
Business-Systems
Successful data extraction
Caterer
Successful data extraction
Chamber-of-Commerce
Successful data extraction
Cleaning-Service
Successful data extraction
Commercial-Appliance-Repair
Successful data extraction
Commerci

In [9]:
df= pd.DataFrame(all_data)

In [10]:
df

name  \
0           Raymond A. Oppel, CPA, LLC   
1               Rodney P. Kaufman, CPA   
2    Apicella Testa & Company PC CPA's   
3         CLA (CliftonLarsonAllen LLP)   
4                The Tic & Tie Company   
..                                 ...   
419    Healing Arts & Physical Therapy   
420                      BBX Nutrition   
421                 Cindy Fox Coaching   
422               Winsupply of Shelton   
423          Troy Industrial Solutions   

                                                  info  \
0      259 Coram AvenueShelton, CT 06484  203-929-6616   
1    569 Boston Post RoadOrange, CT 06497  203-795-...   
2    680 Bridgeport AvenueShelton, CT 06484  (203) ...   
3    2 Enterprise DriveShelton, CT 06484  203-944-2100   
4      55 Merritt BlvdTrumbull, CT 06611  203-212-8557   
..                                                 ...   
419    6 Barn Hill LaneSeymour, CT 06483  630-702-9063   
420   65 Birchbank RoadShelton, CT 06484  203-913-8428   
421  8 Huntington StreetShelton, CT 06484  203-610-...   
422   14 Forest ParkwayShelton, CT 06484  203-929-6344   
423     30 Maple StreetAnsonia, CT 06401  203-734-1695   

                                                social  
0                                                   []  
1                                                   []  
2                                                   []  
3                                                   []  
4                                                   []  
..                                                 ...  
419                                                 []  
420  [https://www.facebook.com/BBXNutrition, https:...  
421  [https://www.facebook.com/findingfreedomwithfo...  
422                                                 []  
423  [https://www.linkedin.com/company/troy-industr...  

[424 rows x 3 columns]

## **Data Manipulation Using Pandas**

In [11]:
df['location']=df["info"].str.split(',').str[0]
df['phone']=df['info'].str.split().str[-1]
df['postal_code']=df['info'].str.split().str[-2]

In [12]:
df

name  \
0           Raymond A. Oppel, CPA, LLC   
1               Rodney P. Kaufman, CPA   
2    Apicella Testa & Company PC CPA's   
3         CLA (CliftonLarsonAllen LLP)   
4                The Tic & Tie Company   
..                                 ...   
419    Healing Arts & Physical Therapy   
420                      BBX Nutrition   
421                 Cindy Fox Coaching   
422               Winsupply of Shelton   
423          Troy Industrial Solutions   

                                                  info  \
0      259 Coram AvenueShelton, CT 06484  203-929-6616   
1    569 Boston Post RoadOrange, CT 06497  203-795-...   
2    680 Bridgeport AvenueShelton, CT 06484  (203) ...   
3    2 Enterprise DriveShelton, CT 06484  203-944-2100   
4      55 Merritt BlvdTrumbull, CT 06611  203-212-8557   
..                                                 ...   
419    6 Barn Hill LaneSeymour, CT 06483  630-702-9063   
420   65 Birchbank RoadShelton, CT 06484  203-913-8428   
421  8 Huntington StreetShelton, CT 06484  203-610-...   
422   14 Forest ParkwayShelton, CT 06484  203-929-6344   
423     30 Maple StreetAnsonia, CT 06401  203-734-1695   

                                                social  \
0                                                   []   
1                                                   []   
2                                                   []   
3                                                   []   
4                                                   []   
..                                                 ...   
419                                                 []   
420  [https://www.facebook.com/BBXNutrition, https:...   
421  [https://www.facebook.com/findingfreedomwithfo...   
422                                                 []   
423  [https://www.linkedin.com/company/troy-industr...   

                         location         phone postal_code  
0         259 Coram AvenueShelton  203-929-6616       06484  
1      569 Boston Post RoadOrange  203-795-1272       06497  
2    680 Bridgeport AvenueShelton      925-9494       (203)  
3       2 Enterprise DriveShelton  203-944-2100       06484  
4         55 Merritt BlvdTrumbull  203-212-8557       06611  
..                            ...           ...         ...  
419       6 Barn Hill LaneSeymour  630-702-9063       06483  
420      65 Birchbank RoadShelton  203-913-8428       06484  
421    8 Huntington StreetShelton  203-610-3673       06484  
422      14 Forest ParkwayShelton  203-929-6344       06484  
423        30 Maple StreetAnsonia  203-734-1695       06401  

[424 rows x 6 columns]